ntlm.go

```diff
@@ -198,6 +217,84 @@ func ntlmSessionResponse(clientNonce [8]byte, serverChallenge [8]byte, password
 	return response(hash, passwordHash)
 }
 
+func ntlmHashNoPadding(val string) []byte {
+	hash := make([]byte, 16)
+	h := md4.New()
+	h.Write(utf16le(val))
+	h.Sum(hash[:0])
+
+	return hash
+}
+
+func hmacMD5(passwordHash, data []byte) []byte {
+	hmacEntity := hmac.New(md5.New, passwordHash)
+	hmacEntity.Write(data)
+
+	return hmacEntity.Sum(nil)
+}
+
+func getTargetInformationData(bytes []byte) (infoBytes []byte, target string, err error) {
```

Comment:
rename func to something more descriptive. No "data" include subject "ntlmv2" or similar or even more specific.
Don't make var name "bytes". You could call it payload, or even data, but not bytes.

Don't call that infoBytes, just info would be fine.
A comment on some of these messages would be helpful, especially references to a NTLMv2 specification.

Reply:
I agree it is kind of a generic name, but the name of the function is chosen as a reference to the related part of the specification. Does that change your mind about the function name?